src/input/pointer.js

```diff
@@ -572,25 +643,28 @@
         if (pointerEventList !== activeEventList) {
             eventType = activeEventList[pointerEventList.indexOf(eventType)];
         }
-
+        
+        // allocated a GUID value if not present
+        if (!rect.GUID) {
+            rect.GUID = me.utils.createGUID();
+        }
+        
         // register the event
-        if (!evtHandlers[eventType]) {
-            evtHandlers[eventType] = [];
+        if (!evtHandlers[rect.GUID]) {
+            evtHandlers[rect.GUID] = {
+                rect : rect,
```

Comment:
actually we should maybe change this part, change the function signature and replace the `rect` parameter by any object implementing the `getBounds()` function (and stores the object reference instead of the rect) and of course change any use of rect by a call to the object `getBounds()` function. 

Then If a `me.Rect` is provided, the behavior will probably be the same, but then if we pass an entity, it will allow to provide dynamic mouse detection for entities with changing shapes.